CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+cmake_minimum_required(VERSION 3.15)
+project(TheKissOfShame VERSION "${ProjectVersion}")
+
+# project description stuff
+set(BaseTargetName TheKissOfShame)
+set(PluginName "The Kiss Of Shame")
+set(ProjectVersion 1.0)
+
+
+include (FetchContent)
+Set(FETCHCONTENT_QUIET FALSE) # shows detailed info when fetching from github
+
+# attempt to build with multiple threads
+set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
+set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+
+# apple universal builds
+if(APPLE)
+set (CMAKE_OSX_DEPLOYMENT_TARGET "11.0" CACHE STRING "Minimum OS X deployment version" FORCE)
+set (CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE INTERNAL "")
+endif()
+
+FetchContent_Declare(
+	JUCE
+	GIT_REPOSITORY https://github.com/juce-framework/JUCE.git
+	GIT_TAG 7.0.11
+	GIT_PROGRESS TRUE
+    GIT_SHALLOW 1
+)
+
+FetchContent_MakeAvailable(JUCE)
+
+# determine folder to place builds, if COPY_PLUGIN_AFTER_BUILD is enabled
+if(APPLE)
+    set(buildFolder "Builds/MacOSX")
+else()
+    if(UNIX)
+        set(buildFolder "Builds/Linux")
+    else()
+        set(buildFolder "Builds/Windows")
+    endif()
+endif()
```

Comment:
Why not use the default platform location?

Reply:
i took most of this cmake from one of my own repos which is configured for github actions, so this would have made it easy to take artifacts out. Left it in there in case it ever gets used (doesn't right now because `COPY_PLUGIN_AFTER_BUILD` is false).

By default platform location, do you mean the default locations where VSTs are installed, or just having COPY_PLUGIN_AFTER_BUILD enabled but without specifying directories?

Reply:
Yeah, so if someone builds it on Windows, it automatically gets copied to `C:\Program Files\Common Files\VST3`. It's not really a big deal, we could add instructions on how to manually copy it to the README. I just find it convenient during development.

Reply:
That makes sense.

I've removed the custom build folder definitions (they can be added back later down the line) and enabled COPY_PLUGIN_AFTER_BUILD, which seems to be installing the plugin to the default plugin folders. (Only tested this on WIndows)